docs/docs/guides/computed-attributes.mdx

```diff
@@ -0,0 +1,265 @@
+
+Once completed, you can load this schema into Infrahub.
+
+:::note
+
+You can specify a transform that doesn't yet exist in Infrahub, attribute will just remain empty. Once the transform is loaded, Infrahub will automatically catch up and compute the value of this attribute.
+
+:::
+
+### Create the Python transform
+
+Please refer to the [Python transform guide](/guides/python-transform) for further details.
+
+1. First we prepare a GraphQL query that returns the data we need
+
+:::warning
+
+GraphQL query used to compute attribute's value requires `ID` as parameter.
+
+```graphql
+query MyQuery($id: ID!) {
+  MyNode(ids: [$id]) {
+    # ...
+  }
+}
+```
+
+:::
+
+```graphql title="computed_circuit_description.gql"
+query CircuitDescriptionQuery($id: ID!) {
+  InfraCircuit(ids: [$id]) {
+    edges {
+      node {
+        circuit_id {
+          value
+        }
+        endpoints {
+          edges {
+            node {
+              name { value }
+              location {
+                node {
+                  name {
+                    value
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+2. Next, the Python script that implements the logic
+
+```python title="computed_circuit_description.py"
+from infrahub_sdk.transforms import InfrahubTransform
+
+
+class ComputedCircuitDescription(InfrahubTransform):
+    query = "computed_circuit_description"
+    url = "computed_circuit_description"
+
+    async def transform(self, data):
+        circuit_dict: dict = data["InfraCircuit"]["edges"][0]["node"]
+
+        detailed_endpoints: list[str] = []
+
+        for endpoint in circuit_dict["endpoints"]["edges"]:
+            detailed_endpoints.append(
+                f'{endpoint["node"]["location"]["node"]["name"]["value"]}::{endpoint["node"]["name"]["value"]}'
+            )
+
+        return f' <- {circuit_dict["circuit_id"]["value"]} -> '.join(detailed_endpoints)
+```
+
+:::note
+
+Make sure that the script returns a string!
+
+:::
+
+3. Tie everything together into `.infrahub.yml` file
+
+```yaml title=".infrahub.yml"
+---
+python_transforms:
+  - name: computed_circuit_description
+    class_name: ComputedCircuitDescription
+    file_path: computed_circuit_description.py
+
+queries:
+  - name: computed_circuit_description
+    file_path: computed_circuit_description.gql
+```
+
+At this stage you should be able to test your transform using `infrahubctl`
+
+```shell title="❯ infrahubctl transform computed_circuit_description id=here_goes_a_circuit_id"
+"site1::endpoint-a <- circuit1 -> site2::endpoint-b"
+```
+
+4. Load the transform in Infrahub
+
+You need to commit these files to a Git repository and link the repository to Infrahub. In order to avoid having the same instructions over and over please refer to the guide [adding a repository to Infrahub](/guides/repository) in order to sync the repository you created and make it accessible within Infrahub.
```

Comment:
You need to commit these files to a Git repository and link the repository to Infrahub. Please refer to the [adding a repository to Infrahub guide](/guides/repository).